### PR TITLE
Refactor statusPercents for lesson progress summary cell

### PR DESCRIPTION
**Pull request by maureensturgeon**
Replace `attempted` with `isStarted` since that's what it's used for. Update the name of the object from statusPercents to studentLessonProgressType

## Links

- [spec]()
- [jira]()

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/progress/progressHelpers.js [hunk 1]
```diff
@@ -126,13 +126,13 @@ export function stageIsAllAssessment(levels) {
  * @param {{id:studentLevelProgressType}} studentProgress An object keyed by
  * level id containing objects representing the student's progress in that level
  * @param {levelType[]} levels An array of levels
- * @returns {object} An object with a percentage of levels in each of the
- * following buckets: attempted, imperfect, completed, incomplete.
+ * @returns {studentLessonProgressType} An object representing student's progress
```
Comment (maureensturgeon) [inline apps/src/templates/progress/progressHelpers.js:129]: Should this be called `studentLessonProgressType` or `studentLessonStatusType`?

File: apps/src/templates/progress/progressTypes.js [hunk 2]
```diff
@@ -33,7 +33,6 @@ const levelWithoutStatusShape = {
   isUnplugged: PropTypes.bool,
   isCurrentLevel: PropTypes.bool,
   isConceptLevel: PropTypes.bool,
-  kind: PropTypes.string,
```
Comment (cforkish) [inline apps/src/templates/progress/progressTypes.js:36]: please don't remove this!

File: apps/src/templates/progress/progressHelpers.js (hunk 1 above)
Comment (cforkish) [reply apps/src/templates/progress/progressHelpers.js:129]: `studentLessonProgressType` -- `status` is a property on the "progress type" object

File: apps/src/templates/sectionProgress/progressTables/ProgressTableSummaryCell.jsx [hunk 3]
```diff
@@ -15,7 +16,7 @@ const styles = {
 export default class ProgressTableSummaryCell extends React.Component {
   static propTypes = {
     studentId: PropTypes.number.isRequired,
-    statusPercents: PropTypes.object,
+    lessonStatus: studentLessonProgressType.isRequired,
```
Comment (cforkish) [inline apps/src/templates/sectionProgress/progressTables/ProgressTableSummaryCell.jsx]: to be consistent with how we use `studentLevelProgressType`, could you rename this prop to `studentLessonProgress`?

File: apps/src/templates/progress/progressHelpers.js [hunk 4]
```diff
@@ -126,13 +126,13 @@ export function stageIsAllAssessment(levels) {
...
+ * in the lesson
  *
  * Note: this function will replace `summarizeProgressInStage` below once we
  * refactor the legacy StudentProgressSummaryCell component
  */
-export function statusPercentsForLevels(studentProgress, levels) {
+export function statusForLesson(studentProgress, levels) {
```
Comment (cforkish) [inline apps/src/templates/progress/progressHelpers.js]: this should be `progressForLesson`. 

now that we have two progress shapes (lesson and level), i'd also rename the `studentProgress` param to `studentLevelProgress` for clarity.

**Review by cforkish — CHANGES_REQUESTED** [review]
this reads much better, thanks!

i made a few renaming requests. going forward, i want our names to be more precise and consistent. "status" has a specific meaning -- it is a string enum refering to the status of a student's progress _on a level_, and lives in `studentLevelProgressType` (going forward. in `progressRedux`-related code it still lives in the `levelType`). atm, lessons don't have a status. their progress type consists of the properties you defined. in other words, "status" is a specific string value, whereas "progress" is a meta object with multiple properties related to student progress. does that make sense?

fwiw, i had the same instinct as you -- inititally `studentLevelProgressType` was called `studentLevelStatusType`. but when i found myself writing a lot of `status.status` code i realized we had a naming collision between the object and the property so i switched to calling the object "progress", which in retrospect i think makes more semantic sense anyway. lmk if you want to discuss this though -- it's important that we all have a consistent and intuitive mental model of what our names mean, so if this one doesn't jive with you i think it's worth discussing.

File: apps/src/templates/progress/progressTypes.js (hunk 2 above)
Comment (cforkish) [reply apps/src/templates/progress/progressTypes.js:36]: nm, thank you for removing this!

**Review by cforkish — APPROVED** [review]
LGTM! this is some really nice cleanup of the original work (@jamescodeorg's comments often have that effect!)

**Pull request merged**